### PR TITLE
Refactor/add sad paths to tests

### PR DESCRIPTION
**Pull request by Eleanorgruth**
Summary of the work done:

- Add sad paths to `RecipeRepository` for the filterTag and filterName methods

This branch includes:

- I also changed the double quotes to single quotes

Contributed to code:

- Eleanor

Needs to review:

- Kiko and Jenn

Additional notes about any issues/questions remaining around this code:

- Can you double-check to make sure that all sad paths are tested?

### CODE REVIEW

File: test/Recipe-test.js [hunk 1]
```diff
@@ -23,35 +23,35 @@ describe('Recipe', () => {
     expect(recipe1.id).to.equal(595736)
```
Comment (knpausch) [inline test/Recipe-test.js:23]: This comment is for line 13 (for some reason github won't let me add it there). 
I know we keep dancing back and forth on using sampleRecipeData[0] or the full hardcoded data. I see a benefit now if we included the full typed out, it would allow us to see the full data and make accurate comparisons to its expect statements instead of us having to refer to the sampleReciepeData file itself. I know it've been changed before but I would recommend, putting the entire recipe here so we can see what we're comparing it to.

File: test/RecipeRepository-test.js [hunk 2]
```diff
@@ -28,11 +28,19 @@ describe('RecipeRepo', () => {
     expect(recipeRepository.createRecipes(sampleRecipeData)).to.deep.equal([recipe1, recipe2])
   })
   it('Should filter a list of recipes based on a tag.', () => {
-    expect(recipeRepository.filterTag("lunch")).to.deep.equal([recipe2])
-    expect(recipeRepository.filterTag("hor d'oeuvre")).to.deep.equal([recipe1])
+    expect(recipeRepository.filterTag('lunch')).to.deep.equal([recipe2])
+    expect(recipeRepository.filterTag('hor d\'oeuvre')).to.deep.equal([recipe1])
+  })
+  it('Should return an empty array if the tag passed in does not match any recipes.', () => {
```
Comment (knpausch) [inline test/RecipeRepository-test.js:34]: good sad path 👍

File: test/RecipeRepository-test.js [hunk 3]
```diff
@@ -28,11 +28,19 @@ describe('RecipeRepo', () => {
...
+    expect(recipeRepository.filterTag('food')).to.deep.equal([])
   })
   it('Should filter list of recipes based on its name.', () => {
-    expect(recipeRepository.filterName("maple dijon apple cider grilled pork chops")).to.deep.equal([recipe2])
-    expect(recipeRepository.filterName("loaded chocolate chip pudding cookie cups")).to.deep.equal([recipe1])
+    expect(recipeRepository.filterName('maple dijon apple cider grilled pork chops')).to.deep.equal([recipe2])
+    expect(recipeRepository.filterName('loaded chocolate chip pudding cookie cups')).to.deep.equal([recipe1])
+  })
+  it('Should return an empty array if the name searched does not match any recipes.', () => {
+    expect(recipeRepository.filterName('calamari')).to.deep.equal([])
```
Comment (knpausch) [inline test/RecipeRepository-test.js:43]: good sad path. How about adding a test if an empty string was inputted?

**Review by knpausch — CHANGES_REQUESTED** [review]
thnx for formatting the code to single quotes. I added 2 comments to incorporate if you agree

File: test/Recipe-test.js (hunk 1 above)
Comment (Eleanorgruth) [reply test/Recipe-test.js:23]: I would argue to keep it as it is, I think that if we do make that change we wouldn't be using the sample data at all in this test. Heather didn't have feedback on this and said "Good job creating a sample data file for your testing!"

File: test/RecipeRepository-test.js (hunk 3 above)
Comment (Eleanorgruth) [reply test/RecipeRepository-test.js:43]: Good call, I'm adding that!
